kyuubi-server/web-ui/README.md

```diff
@@ -0,0 +1,95 @@
+<!--
+ - Licensed to the Apache Software Foundation (ASF) under one or more
+ - contributor license agreements.  See the NOTICE file distributed with
+ - this work for additional information regarding copyright ownership.
+ - The ASF licenses this file to You under the Apache License, Version 2.0
+ - (the "License"); you may not use this file except in compliance with
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+# Apache Kyuubi(incubating) Web UI
+
+The web ui listens at `http://localhost:10009`.
+
+The web ui is work in progress. Kyuubi server starts an HTTP server (by default at port 10009)
+that serves the new web pages and additional background requests.
+
+## Server Backend
+
+The server side of the ui is implemented using [Jetty](https://www.eclipse.org/jetty/) 
+for REST paths.
+The framework has very lightweight dependencies.
+
+## UI Frontend 
+
+The web ui is implemented using *Angular*. The ui build infrastructure uses *node.js*.
+
+### Preparing the Build Environment
+
+Depending on your version of Linux, Windows or MacOS, you may need to manually install *node.js*
+
+#### Ubuntu Linux
+
+Install *node.js* by following [these instructions](https://nodejs.org/en/download/).
+
+Verify that the installed version is at least *10.9.0*, via `node --version`.
+
+#### MacOS
+
+First install *brew* by following [these instructions](http://brew.sh/).
+
+Install *node.js* via:
+
+```
+brew install node
+```
+
+### Building
+
+The build process downloads all requires libraries via the *node.js* package management tool (*npm*)
+The final build tool is *@angular/cli*.
+
+```
+cd kyuubi-server/web-ui
+npm install
```

Comment:
"7 vulnerabilities (4 moderate, 2 high, 1 critical)"
does it matter?

Reply:
Where or how to show this information?

Reply:
simple run this command

Reply:
Why it does not show this warning in my local? Full information after I invoked:

```
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@es-joy/jsdoccomment@0.17.0',
npm WARN EBADENGINE   required: { node: '^12 || ^14 || ^16 || ^17' },
npm WARN EBADENGINE   current: { node: 'v18.0.0', npm: '8.6.0' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'eslint-plugin-jsdoc@37.6.1',
npm WARN EBADENGINE   required: { node: '^12 || ^14 || ^16 || ^17' },
npm WARN EBADENGINE   current: { node: 'v18.0.0', npm: '8.6.0' }
npm WARN EBADENGINE }

> kyuubi-web-ui@2.0.0 prepare
> cd ../.. && husky install kyuubi-server/web-ui/.husky

husky - Git hooks installed

removed 1 package in 2s

160 packages are looking for funding
  run `npm fund` for details
```

Reply:
I think it is caused by different Node.js versions. I'm using Node.js 16.